app/models/product_user_creator.rb

```diff
@@ -0,0 +1,21 @@
+class ProductUserCreator
+  def self.create(user:, product:, approver:)
+    product_user = ProductUser.new(
+      product: product,
+      user: user,
+      approved_by: approver.id,
+      approved_at: Time.zone.now,
+    )
+
+    product_user.transaction do
+      product_user.save && manage_training_request(product_user)
```

Comment:
In order for the transaction to work, wouldn't the `save` and `destroy` need to be the bang versions? Otherwise they just return true/false, which I don't think `transaction` responds to.